addons/knobs/src/html/registerKnobs.js

```diff
@@ -0,0 +1,60 @@
+import addons from '@storybook/addons';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { subscriptionsStore } from '@storybook/core/client';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { forceReRender } from '@storybook/html';
+import { manager } from '../base';
+
+const channel = addons.getChannel();
```

Comment:
channel is set at some point in runtime, so it's unsafe to call it at module top level. You can just inline this call at each usage place, because all that it does is return a field on `addons` object

Reply:
Like this:

```
function disconnectCallbacks() {
  const channel = addons.getChannel()
  channel.removeListener('addon:knobs:knobChange', knobChanged);
  channel.removeListener('addon:knobs:knobClick', knobClicked);
  channel.removeListener('addon:knobs:reset', resetKnobs);
  knobStore.unsubscribe(setPaneKnobs);
}
```